refresh/equaler.go

```diff
@@ -0,0 +1,5 @@
+package refresh
+
+type Equaler[T any] interface {
+	Equal(T) bool
```

Comment:
I think convention is `Equals`, and you may not need to export this interface?

Reply:
Un-exported the interface. Regarding the function name, it seems go-cmp looks for `Equal`, so I was using that as a guide (though we aren't really using go-cmp itself of course) -
![image](https://github.com/user-attachments/assets/6f8a45c4-c708-4dae-bee5-15d43e15984f)

Reply:
The standard library convention is Equal, which is also probably why `cmp` uses it:

- `net.(IP).Equal`: https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/net/ip.go;l=359
- `time.(Time).Equal`: https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/time/time.go;l=301